File: Java/Logic-2/closeFar.java

public class logic2 {
	
	// given three ints, a b c, return true if one of b or c is "close" (differing from a by at most 1)
	// while the other is "far", differing from both other values by 2 or more 
	public boolean closeFar(int a, int b, int c) {
		
		boolean bool = false;
		int close; 
		int far;
		
		if (Math.abs(b - a) < Math.abs(c - a)) {
			close = b;
			far = c;
		}
		else {
			close = c;
			far = b;
		}
		
		if (Math.abs(a - close) < 2 && (Math.abs(far - close) > 1 && Math.abs(far - a) > 1))
			bool = true;
		
		return bool;
		
	}
	
	public static void main(String[] args) {
		
		logic2 o = new logic2();
		
		System.out.println(o.closeFar(1, 2, 10));
		System.out.println(o.closeFar(1, 2, 3));
		System.out.println(o.closeFar(4, 1, 3));
		
	}

}

File: Python/Logic-2/close_far.py

# given three ints, a b c, return true if one of b or c is "close" (differing from a by at most 1)
# while the other is "far", differing from both other values by 2 or more 
def close_far(a, b, c):
    bool = False

    if abs(b - a) < abs(c - a):
        close = b
        far = c
    else:
        close = c
        far = b

    if abs(close - a) < 2 and abs(far - close) > 1 and abs(far - a) > 1:
        bool = True

    return bool

print(close_far(1, 2, 10))
print(close_far(1, 2, 3))
print(close_far(4, 1, 3))
